packages/webapp/pages/notifications.tsx

```diff
@@ -91,6 +134,73 @@ const Notifications = (): ReactElement => {
 
   usePromotionModal();
 
+  const { show } = useContextMenu({ id: contextId });
+  const [notification, setNotification] = useState<Notification>();
+  const {
+    preferences,
+    isFetching,
+    clearNotificationPreference,
+    muteNotification,
+  } = useNotificationPreference({
+    params: notification
+      ? [
+          {
+            type: notificationPreferenceMap[notification.type],
+            referenceId: notification.referenceId,
+          },
+        ]
+      : [],
+  });
+
+  const onOptionsClick = (e: React.MouseEvent, item: Notification) => {
+    e.preventDefault();
+
+    if (notification) {
+      const { referenceId, type } = notification;
+      const clickedSameButton =
+        referenceId === item.referenceId && type === item.type;
+      setNotification(clickedSameButton ? undefined : item);
+      return;
+    }
+
+    setNotification(item);
+    const { right, bottom } = e.currentTarget.getBoundingClientRect();
+    show(e, { position: { x: right, y: bottom + 4 } });
+  };
+
+  const getIcon = () => {
+    if (isFetching) return <Loader />;
+
+    return preferences[0]?.status === NotificationPreferenceStatus.Muted ? (
+      <BellIcon />
+    ) : (
+      <BellDisabledIcon />
+    );
+  };
```

Comment:
More React way is I thin calling it `Icon` and then just rendering below like `<Icon />`, same for `getCopy` below.